File: LearnAndroid/Day6/app/src/main/java/com/example/day6/beans/Item.java
package com.example.day6.beans;

import android.graphics.drawable.Drawable;
import android.net.Uri;

import com.chad.library.adapter.base.entity.MultiItemEntity;

/**
 * @author wellorbetter
 */

public class Item implements MultiItemEntity {
    public static final int TYPE_IMAGE = 1;
    public static final int TYPE_TEXT = 2;

    private final int mItemType;
    private String mText;
    private Uri mImage;
    private boolean isLike;

    public Item(int itemType, String text) {
        this.mItemType = itemType;
        this.mText = text;
    }

    public Item(int itemType, Uri image) {
        this.mItemType = itemType;
        this.mImage = image;
        this.isLike = false;
    }

    @Override
    public int getItemType() {
        return mItemType;
    }

    public String getText() {
        return mText;
    }

    public Uri getImage() {
        return mImage;
    }

    public boolean isLike() {
        return isLike;
    }

    public void setLike(boolean like) {
        isLike = like;
    }
}

File: LearnAndroid/Day11/app/src/main/java/com/example/day11/bean/Item.kt
package com.example.day11.bean

import android.graphics.drawable.Drawable
import com.chad.library.adapter.base.entity.MultiItemEntity

/**
 * @author wellorbetter
 */
class Item(val img: Drawable? = null, var name: String = "", var id: Int = 0, override val itemType: Int): MultiItemEntity {
    var poetName:String = ""
    var poemName:String = ""
    var poemContent:String=""
    var like:Boolean=false
    constructor(itemType: Int):this(null, "", 0, POEM_TEXT)
    constructor(poetName:String, poemName:String, poemContent:String, itemType: Int) : this(itemType) {
        this.poetName = poetName
        this.poemName = poemName
        this.poemContent = poemContent
    }
    companion object {
        const val PERSON_CARD = 0
        const val POEM_TEXT = 1

        // 这是一个扩展函数，用于判断Item实例是否是PERSON_CARD类型
        fun Item.isPersonCard() = this.itemType == PERSON_CARD

        fun Item.isPoemText() = this.itemType == POEM_TEXT

    }

    // 这里可以添加其他Item类的方法或属性
}